src/main/java/com/redhat/devtools/intellij/tektoncd/listener/SaveInEditorListener.java

```diff
@@ -143,5 +153,27 @@ public boolean maySaveDocument(@NotNull Document document, boolean isSaveExplici
         Notifications.Bus.notify(notification);
         return false;
     }
+
+    private boolean isFileToPush(Project project, Document document, VirtualFile vf) {
+        Editor selectedEditor = FileEditorManager.getInstance(project).getSelectedTextEditor();
+        // if file is not the one selected, skip it
+        if (selectedEditor.getDocument() != document) return false;
+        // if file is not related to tekton, skip it
+        if (vf == null || vf.getUserData(KIND_PLURAL) == null || vf.getUserData(KIND_PLURAL).isEmpty()) {
+            return false;
+        }
+        return true;
+    }
+
+    private Project getProject() {
```

Comment:
There is something buggy here: if you have 2 different projects (windows) opened in IJ, you will get the wrong project and then detect that push is not to be performed. Remember that the Tekton window is attached to a project because the factory is given the Project and then Project can be retrieved when we execute our action as should be stored in the context as well.